fe/src/main/java/org/apache/doris/analysis/CastExpr.java

```diff
@@ -248,4 +248,33 @@ public boolean canHashPartition() {
         return false;
     }
 
+    @Override
+    public Expr getResultValue() throws AnalysisException {
+        final Expr value = children.get(0).getResultValue();
+        if (!(value instanceof LiteralExpr)) {
+            return this;
+        }
+
+        final Expr targetExpr = castTo((LiteralExpr)value);
+        return targetExpr;
+    }
+
+    private Expr castTo(LiteralExpr value) throws AnalysisException {
```

Comment:
use `LiteralExpr.uncheckCast`

Reply:
This can't use uncheckCast, it is used by impliclit cast and it will return new CastExpr for no supporting type, so it will cause dead loop.